pkg/generators/snippets/setters.go

```diff
@@ -7,73 +7,89 @@ import (
 	"k8s.io/gengo/types"
 )
 
-func GenerateSetterForPrimitiveType(root *types.Type, member types.Member) (string, generator.Args) {
+func GenerateSetterForPrimitiveType(root *types.Type, parent *types.Type, member types.Member) (string, generator.Args) {
 	args := generator.Args{
-		"type":       root,
-		"funcName":   funcName(member),
-		"memberName": member.Name,
-		"memberType": member.Type,
+		"type":           root,
+		"funcName":       funcName(member),
+		"memberAccessor": member.Name,
+		"memberType":     member.Type,
+	}
+
+	if root != parent {
+		args["memberAccessor"] = fmt.Sprintf("%s.%s", parent.Name.Name, member.Name)
 	}
```

Comment:
This can be refactored into a function that returns the memberAccessor given the root, parent and member.